ci/docker/java-jni-manylinux-201x.dockerfile

```diff
@@ -52,4 +52,5 @@ RUN PYTHON=python /arrow/ci/scripts/install_gcs_testbench.sh default
 ENV ARROW_HOME=/tmp/local \
     ARROW_JAVA_CDATA=ON \
     ARROW_JAVA_JNI=ON \
-    ARROW_USE_CCACHE=ON
+    ARROW_USE_CCACHE=ON \
+    ARROW_DATASET=ON
```

Comment:
Could you keep this list in alphabetical order?

Reply:
@kou here we don't need the `ARROW_DATASET=ON`, right?

Reply:
Right.

How about the following?

```diff
diff --git a/ci/scripts/java_test.sh b/ci/scripts/java_test.sh
index dd483ff254..5efda4318f 100755
--- a/ci/scripts/java_test.sh
+++ b/ci/scripts/java_test.sh
@@ -38,14 +38,10 @@ pushd ${source_dir}
 ${mvn} clean test
 
 projects=()
-if [ "${ARROW_DATASET}" = "ON" ]; then
-  projects+=(gandiva)
-fi
-if [ "${ARROW_GANDIVA}" = "ON" ]; then
-  projects+=(gandiva)
-fi
-if [ "${ARROW_ORC}" = "ON" ]; then
+if [ "${ARROW_JAVA_JNI}" = "ON" ]; then
   projects+=(adapter/orc)
+  projects+=(dataset)
+  projects+=(gandiva)
 fi
 if [ "${#projects[@]}" -gt 0 ]; then
   ${mvn} clean test \
```

Reply:
Perfect!!!

Reply:
I will add this. Thanks @kou

Reply:
@kou I added the suggestion and started the crossbows.